File: insee-promotion-admin-client/app/components/enum/GiftStatus.js
export class  GiftStatus {
    constructor(status) {
        this.status = status
    }

    getStatus() {
        return this.status;
    }

    static findByStatus(status) {
        switch(status) {
            case 1 : return WAITING_RECEIVE;
            case 2 : return RECEIVED;
            case 3 : return ROLLED;
            case 4 : return WAITING_SENT;
        }
    }

    static getColor(status) {
        switch(status) {
            case 1: return '#1890ff';
            case 2: return '#b71c1c';
            case 3: return  '#ff4d4f';
            case 4: return '#ff4d4f'
        }
    }

    static getName(status) {
        switch(status) {
            case 1 : return "Chờ nhận"
            case 2 : return "Đã nhận"
            case 3 : return "Đã quay"
            case 4 : return "Chờ Gửi"
        }
    }
    
}
export const WAITING_RECEIVE = new GiftStatus(1);
export const RECEIVED = new GiftStatus(2);
export const ROLLED = new GiftStatus(3)
export const WAITING_SENT = new GiftStatus(4);

File: ZaloBotServer/src/main/java/com/vn/insee/zalobot/common/status/GiftStatus.java
package com.vn.insee.zalobot.common.status;

public enum GiftStatus {
    SEND(1), RECEIVED(2), ROLLED(3), APPROVED(4), WAITING_SENT(5);
    private int status;

    GiftStatus(int status) {
        this.status = status;
    }

    public int getStatus() {
        return status;
    }

    public static GiftStatus findByValue(int status) {
        switch (status) {
            case 1 : return SEND;
            case 2 : return RECEIVED;
            case 3 : return ROLLED;
            case 4 : return APPROVED;
            case 5 : return WAITING_SENT;
        }
        return null;
    }
}
